hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/interceptor/EnhancementAsProxyLazinessInterceptor.java

```diff
@@ -59,6 +61,9 @@ public EnhancementAsProxyLazinessInterceptor(
 		final EntityPersister entityPersister = session.getFactory().getMetamodel().entityPersister( entityName );
 		this.inLineDirtyChecking = entityPersister.getEntityMode() == EntityMode.POJO
 				&& SelfDirtinessTracker.class.isAssignableFrom( entityPersister.getMappedClass() );
+		// if self-dirty tracking is enabled but DynamicUpdate is not enabled then we need to initialise the entity
+		// 	because the pre-computed update statement contains even not dirty properties and so we need all the values
+		initializeOnRead = !inLineDirtyChecking || !entityPersister.getEntityMetamodel().isDynamicUpdate();
```

Comment:
IIUC, ```initializeOnRead``` indicates that the entity should be initialized before writing an attribute value to the entity.

If that is the case, the field should be named something like ```initializeBeforeWrite```.

Reply:
thanks @gbadner I have applied your suggestion